company-template.el

```diff
@@ -109,14 +139,22 @@ Leave point at the end of the field."
       (overlay-put ov 'display display))
     (overlay-put ov 'company-template-parent templ)
     (overlay-put ov 'insert-in-front-hooks '(company-template-insert-hook))
+    (when clear-function
+      (overlay-put ov 'clear-function clear-function))
     (push ov siblings)
     (overlay-put templ 'company-template-fields siblings)))
 
 (defun company-template-remove-field (ovl &optional clear)
   (when (overlayp ovl)
     (when (overlay-buffer ovl)
       (when clear
-        (delete-region (overlay-start ovl) (overlay-end ovl)))
+        (let ((clear-function (overlay-get ovl 'clear-function))
+              (start (overlay-start ovl))
+              (end (overlay-end ovl)))
+          (if (and (not (eq clear 'clear-bounds-only))
```

Comment:
This is a minor complaint (don't bother if you're tired of refining this), but maybe the clear-function should just receive the beginning of the next field? Then it could be called simply at the end of `company-template-clear-field`.

Even the beginning of the next field is not necessary for our current purpose, since we can be sure the following comma and spaces are outside of it.